robottelo/rerun_failures/uncollector.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+
+from robottelo.constants import DEFECT_TYPES
+from robottelo.report_portal.portal import Launch
+from robottelo.report_portal.portal import ReportPortal
+
+
+def pytest_addoption(parser):
+    """Add options for pytest to collect only failed tests"""
+    help_script = '''
+        Collects only failed tests in Report Portal to run only failed tests.
+
+        Options: [ all | specific_defect_type | comma_sparated_list_of defect_types ]
+
+        Defect_types:
+        all: This is combine of system_issue, to_investigate and no_defect failed tests.
+        product_bug: Rerun failed tests marked with product_bug on RP
+        rhel_bug: Rerun failed tests marked with rhel_bug on RP
+        satellite_bug: Rerun failed tests marked with rhel_bug on RP
+        automation_bug': Rerun failed tests marked with automation_bug on RP
+        robottelo_bug': Rerun failed tests marked with robottelo_bug on RP
+        nailgun_bug': Rerun failed tests marked with nailgun_bug on RP
+        airgun_bug': Rerun failed tests marked with airgun_bug on RP
+        system_issue': Rerun failed tests marked with system_issue on RP
+        saucelabs_issue': Rerun failed tests marked with saucelabs_issue on RP
+        to_investigate': Rerun failed tests marked with to_investigate on RP
+        no_defect': Rerun failed tests marked with no_defect on RP
+        '''
+    parser.addoption("--only-failed", help=help_script)
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_collection_modifyitems(items, config):
+    """
+
+    """
+    # Argument Sanitization
+    fail_args = config.getvalue('only_failed')
+    if not fail_args:
+        return
+    fail_args = fail_args.split(',').strip() if ',' in fail_args else [fail_args]
+    allowed_args = [*DEFECT_TYPES.keys()]
+    allowed_args.append('all')
+    if not set(fail_args).issubset(set(allowed_args)):
+        pytest.fail(
+            f'Incorrect values to pytest option --only-failed are provided as'
+            f'{fail_args} should be one of {allowed_args}'
+        )
+    # Collect tests from Report Portal and analyze launch to decide on rerun
+    sat_version = '6.7.0'  # To_BE_Calculated from satellite
+    rp = ReportPortal()
+    launch = Launch(rp, sat_version=sat_version)
+    launch_info = launch.launch
+    launch_stats = launch_info['statistics']['executions']
+    pass_percent = round(100 * (int(launch_stats['passed']) / int(launch_stats['total'])))
+    if launch_info['isProcessing']:
+        pytest.fail(f'The latest launch of satellite version {sat_version} is not Finished yet.')
+    if pass_percent < 80:
+        pytest.fail(
```

Comment:
I don't think we should call pytest.fail in either of these situations, as this is not a failing test case.

The hook should return without modifying items (after logging), since the available data is insufficient to modify the test collection.

Reply:
I wanted to somehow halt the pytest going to the next plugin to collect tests else we wont be able to satisfy the purpose of using `--only-failed` option.

If the user won't look at the logging, he/she might think the tests running are being filtered are failed tests. To give you an example, our CI jobs, people won't look at the console logs they only see the test reports.

If they really don't want to care for RP fail tests, they can remove the --only-failed option and rerun the pytest session.

Reply:
Btw, I can raise an exception instead of pytest.fail.